interfaces/cython/cantera/composite.py

```diff
@@ -509,6 +509,9 @@ def __init__(self, phase, shape=(0,), states=None, extra=None):
         if len(self._shape) == 1:
             self._indices = list(range(self._shape[0]))
             self._output_dummy = self._indices
+        elif (len(self._shape) == 0 and len(states) == 0):
```

Comment:
Why did you add the `and len(states) == 0` clause to the check here? Just the check on `self._shape` is sufficient in my testing to resolve the bug this is intended to fix. To be clear, I'm not asking you to delete the `len(states)` check, I'm wondering why it is necessary.

You also do not need the parentheses around the statement.

Reply:
arr[1] would have been treated the same way but we wish to depend on arr
for instance, this part of the test_slicing_onedim should fail.

```
test_slicing_onedim(self):
        states = ct.SolutionArray(self.gas, 5)
        states.TPX = np.linspace(500, 1000, 5), 2e5, 'H2:0.5, O2:0.4'
        T0 = states.T
        H0 = states.enthalpy_mass

        # Verify that original object is updated when slices change
        state = states[1]
        state.TD = 300, 0.5
        self.assertNear(states.T[0], 500)
        self.assertNear(states.T[1], 300)
```
I would remove the Paranthesis.